models/repo.go

```diff
@@ -2363,3 +2401,28 @@ func (repo *Repository) CreateNewBranch(doer *User, oldBranchName, branchName st
 
 	return nil
 }
+
+// CreateNewBranchFromCommit creates a new repository branch
+func (repo *Repository) CreateNewBranchFromCommit(doer *User, commit, branchName string) (err error) {
```

Comment:
should I also move `CreateNewBranch` function that was already there to repo_branch.go? As these functions are doing very similar things I just did not want them to be in different files